2020-05/robot-name/robot-name.js

```diff
@@ -2,46 +2,121 @@
 // provided as a convenience to get your started writing code faster.
 
 export class Robot {
-  constructor(name){
-    this.nameHistory = []
-    this.name = this.randomName()
+  // A list of all robot names used.
+  static nameHistory = {}
+
+  // A private robot name.
+  #name = ''
+
+  constructor() {
+    this.#name = this.newUniqueName()
+  }
+
+  /**
+   * A getter to return the "name" of the current robot.
+   *
+   * @return {string|*}
+   */
+  get name() {
+    return this.#name
   }
 
+  /**
+   * Create the random name for the robot in the format of two uppercase letters
+   * followed by three digits, such as RX837 or BC811.
+   *
+   * @return {string}
+   */
   randomName() {
-    let result = ''
-    result += this.makeid(2)
+    let name = ''
 
-    for (var i = 0; i < 3; i++) {
-      result += Math.floor(Math.random() * 10).toString()
+    // Generate the first two alpha-numeric characters.
+    const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
+    for (let i = 0; i < 2; i++) {
+      name += characters.charAt(Math.floor(Math.random() * characters.length))
     }
-    // console.log("before adding")
-    // console.log(this.nameHistory)
-    this.nameHistory.push(result)
-    // console.log("after adding")
-    // console.log(this.nameHistory)
-    return result
-  }
-
-  makeid(length) {
-    var result = '';
-    var characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
-    var charactersLength = characters.length;
-    for ( var i = 0; i < length; i++ ) {
-    result += characters.charAt(Math.floor(Math.random() * charactersLength));
+
+    // Generate the remaining three random numbers.
+    for (let i = 0; i < 3; i++) {
+      name += Math.floor(Math.random() * 10).toString()
     }
-    return result;
+
+    return name
+  }
+
+  /**
+   * Generate a new Unique name for a robot.
+   *
+   * @return {string}
+   */
+  newUniqueName() {
+    let name = ''
+    do {
+      name = this.randomName()
+    } while (this.notUnique(name))
```

Comment:
I like this logic :+1: - do this, while that... or just keep generating names until you get a unique one.